lib/qbo_api.rb

```diff
@@ -104,40 +114,65 @@ def reconnect
   end
 
   def request(method, path:, entity: nil, payload: nil, params: nil)
-    raw_response = connection.send(method) do |req|
-      path = finalize_path(path, method: method, params: params)
+    raw_response = raw_request(method, conn: connection, path: path, params: params, payload: payload)
+    response(raw_response, entity: entity)
+  end
+
+  def raw_request(method, conn:, path:, payload: nil, params: nil)
+    path = finalize_path(path, method: method, params: params)
+    @last_response = conn.public_send(method) do |req|
       case method
       when :get, :delete
         req.url path
       when :post, :put
         req.url path
         req.body = payload.to_json
+      else raise ArgumentError, "Unhandled request method '#{method.inspect}'"
       end
     end
-    response(raw_response, entity: entity)
+  end
+
+  def last_response
+    return @last_response if defined(@last_response)
   end
 
   def response(resp, entity: nil)
-    data = JSON.parse(resp.body)
-    if entity
-      entity_response(data, entity)
-    else
-      data
-    end
+    data = parse_response_body(resp)
+    entity ? entity_response(data, entity) : data
   rescue => e
-    # Catch fetch key errors and just return JSON
+    # NOTE(BF): Rescue all StandardError and return parsed JSON body
```

Comment:
best view this in split view.  I don't like this timid code which rescues all StandardError.  Related, to JSON requests perhaps we can add the `conn.use FaradayMiddleware::ParseJson, parser_options: { symbolize_names: false }` middleware.

Relatedly, I'd like to wrap json and xml encoding/decoding rather than specifying the library